chatcommunicate.py

```diff
@@ -262,9 +262,9 @@ def tell_rooms(msg, has, hasnt, notify_site="", report_data=None):
 
         if room.block_time < timestamp and _global_block < timestamp:
             if report_data and "delay" in _room_roles and room_id in _room_roles["delay"]:
-                threading.Thread(name="delayed post",
-                                 target=DeletionWatcher.post_message_if_not_deleted,
-                                 args=(msg_pings, room, report_data)).start()
+                task = Tasks.later(_msg_queue.put, (room, msg_pings, report_data), after=300)
+
+                GlobalVars.deletion_watcher.subscribe(report_data[0], callback=task.cancel)
```

Comment:
Should this use `subscribe`'s `timeout` param derived from `room.block_time`? Calling `cancel` on the task is useless after the message has been posted.

Reply:
Ignore this one too, I can't read.